Automate release creation

GabLeRoux opened the pull request:
Quick draft to see how we can create releases out of builds from pipelines

GabLeRoux committed:
Automate release creation with dawidd6/action-download-artifact and fnkr/github-action-ghr

Gcenx commented:
Current on mobile so editing the workflow will be annoying before installing dependencies add `brew update` first that will force upgrade the packages on the GitHub images.

That will ensure the very latest bottles are used

Gcenx committed:
build_monolithic.yml: always run brew update

Github CI images get updated infrequently where we want the latest formula/bottles
